### ISSUE TITLE
Suppressions are not applied on errors reported by plugins

### ISSUE THREAD

**Issue by boesing** [vimeo/psalm#7840]
Hey there,

I have a plugin which reports a `TooFewArguments` issue.
When I try to use `@psalm-suppress` to temporarily suppress that error (as its a bug in the plugin which I haven't fixed yet) but when I do so, the `@psalm-suppress` is kinda - ignored.

So the weird thing is, that I have enabled "find unused psalm suppressions" via config file. But that suppression is not reported by that check. But the error is also not suppressed.

Any ideas what could cause this?

**Comment by orklah** [comment]
I'm not sure Psalm expects to find one of it's own Issue coming from a plugin. There is PluginIssue designed for that, which can be suppressed (at least from issueHandler perspective), not sure about @psalm-suppress though...

**Comment by boesing** [comment]
mh, thats sad since that error actually reflects the exact Problem. but I will see if I can switch to the plugin issue. 
but thats less detailed 😕

**Comment by orklah** [comment]
Note that PluginIssue is made to be inherited (that should have been an interface I think). Feel free to make a TooFewArgumentsPlugin or something like that to be even more detailed than core issue :)

**Comment by boesing** [comment]
I did not had a more specific issue. Its just that psalm does not detect some issues in specific scenarios. The plugin does but does not need to provide more specific information.

Anyways, I've implemented `PluginIssue` now, but not 100% sure if its possible to suppress the concrete issue (as it is actually named the same as the core issue but in a different namespace).

For now, that might be good enough - if not, I'll see why it won't work and create a PR.
